### DISCUSSION TITLE
qemu-guest-kragent for x86 gokrazy-in-QEMU on Proxmox

### DISCUSSION THREAD

**Discussion by bradfitz** [gokrazy/gokrazy#190] [Show and tell]
I just wrote https://github.com/bradfitz/qemu-guest-kragent for my x86 gokrazy VMs running on Proxmox.

Hope anybody else finds it useful too!

Also, some tips: disable USB hotplug and USB tablet/pointer support to save a bunch of CPU. And using the "vga: virtio" graphics card ("virtio-GPU" or something in one UI) then /dev/fb0 works and you can get a framebuffer, but the fbstatus command uses too much CPU so I've stopped using it now that I have `qemu-guest-kragent` to tell me the VM's IP.

![c2ef209a40ba82d7](https://user-images.githubusercontent.com/2621/229333688-0b80da06-f756-4c8d-8a25-863d9f7a007f.png)

And here's Proxmox's summary page of a VM, showing the IPs now (with my IPv6 redacted):

<img width="650" alt="Screenshot 2023-04-01 at 10 37 26 PM" src="https://user-images.githubusercontent.com/2621/229333750-5906d2d7-e214-444e-b74e-117296f11b62.png">

**Comment by damdo** [comment]
Thanks @bradfitz I'm planning to run gokrazy on Proxmox in the future, this is going to be quite useful! :)
